core/backend/src/TxnManager.ts

```diff
@@ -406,6 +483,98 @@ export class TxnManager {
     this.onAfterUndoRedo.raiseEvent(isUndo);
     IpcHost.notifyTxns(this._iModel, "notifyAfterUndoRedo", isUndo);
   }
+  /** @internal */
+  protected _onRebaseTxnBegin(txn: TxnArgs) {
+    this.onRebaseTxnBegin.raiseEvent(txn);
+  }
+
+  /** @internal */
+  protected _onRebaseLTxnEnd(txn: TxnArgs) {
+    this.onRebaseLTxnEnd.raiseEvent(txn);
+  }
+
+  /** @internal */
+  protected _onRebaseLocalTxnConflict(args: RebaseChangesetConflictArgs): DbConflictResolution {
+    try {
+      const resolution = this.changeMergeManager.onConflict(args);
+      if (resolution !== undefined)
+        return resolution;
+    } catch (err) {
+      Logger.logError(BackendLoggerCategory.IModelDb, BentleyError.getErrorMessage(err));
+      return DbConflictResolution.Abort;
+    }
+
+    const category = "DgnCore";
+    const interpretConflictCause = (cause: DbConflictCause) => {
+      switch (cause) {
+        case DbConflictCause.Data:
+          return "data";
+        case DbConflictCause.NotFound:
+          return "not found";
+        case DbConflictCause.Conflict:
+          return "conflict";
+        case DbConflictCause.Constraint:
+          return "constraint";
+        case DbConflictCause.ForeignKey:
+          return "foreign key";
+      }
+    };
+
+    if (args.cause === DbConflictCause.Data && !args.indirect) {
+      if (args.tableName === "be_Prop") {
+        if (args.getValueText(0, DbChangeStage.Old) === "ec_Db" && args.getValueText(1, DbChangeStage.Old) === "localDbInfo") {
+          return DbConflictResolution.Replace;
+        }
+      }
+      if (args.tableName.startsWith("ec_")) {
+        return DbConflictResolution.Skip;
+      }
+      const msg = "UPDATE/DELETE before value do not match with one in db or CASCADE action was triggered.";
+      args.setLastError(msg);
+      Logger.logError(category, msg);
+      args.dump();
+      return DbConflictResolution.Replace;
+    }
+
+    if (args.cause === DbConflictCause.Conflict) {
+      if (args.tableName.startsWith("ec_")) {
+        return DbConflictResolution.Skip;
+      }
+      const msg = "PRIMARY KEY INSERT CONFLICT - rejecting this changeset";
+      args.setLastError(msg);
+      Logger.logError(category, msg);
+      args.dump();
+      return DbConflictResolution.Abort;
+
+    }
+
+    if (args.cause === DbConflictCause.ForeignKey) {
+      const nConflicts = args.getForeignKeyConflicts();
+      const msg = `Detected ${nConflicts} foreign key conflicts in ChangeSet. Aborting merge.`;
+      args.setLastError(msg);
+      return DbConflictResolution.Abort;
+    }
+
+    if (args.cause === DbConflictCause.NotFound) {
+      return DbConflictResolution.Skip;
+    }
+
+    if (args.cause === DbConflictCause.Constraint) {
+      if (Logger.isEnabled(category, LogLevel.Info)) {
```

Comment:
do we really need to wrap log statements to specific levels with checks if that level is enabled?

Reply:
Its copied to produce same errors/warning as in native. To match logs. But in case of rebase we do not need to be similar as its new code. I will update this method to minimally log and use different category.